rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,89 @@
+---
+authors: Andrew LeFevre (andrew.lefevre@goteleport.com)
+state: draft
+---
+
+# RFD 98 - Registered OpenSSH Nodes
+
+## Required approvers
+
+* Engineering: @jakule && @r0mant
+* Product: @klizhentas
+* Security: @reedloden
+
+## What
+
+Allow OpenSSH nodes to be registered in a Teleport cluster.
+
+## Why
+
+[Agentless EC2 discovery mode](https://github.com/gravitational/teleport/issues/17865) will discover and configure OpenSSH nodes so they can authenticate with a cluster. But those OpenSSH nodes aren't registered as `node` resources in the backend. We need a way to register agentless OpenSSH nodes as `node` resources so they can be viewed and managed by users. RBAC and session recording should function correctly with registered OpenSSH nodes as well.
+
+## Details
+
+### OpenSSH CA
+
+For security related reasons that will be discussed below, a new CA will be added called OpenSSH CA. The OpenSSH CA will be responsible for generating and signing user certificates that will be used to authenticate with registered OpenSSH nodes. The OpenSSH CA will not be used to generate certificates for existing or future *unregistered* OpenSSH nodes, preserving backwards compatibility for existing OpenSSH nodes.
+
+### RBAC
+
+When OpenSSH nodes are registered currently, RBAC checks for those nodes are not preformed. Even setting `auth_service.session_recording` to `proxy` in an Auth Server's config file does not help. RBAC logic will have to be updated so RBAC checks for registered OpenSSH nodes are preformed.
+
+When `(lib/srv.AuthHandlers).UserKeyAuth` is called to authenticate a node's user certificate, it will check what CA signed the certificate. If the OpenSSH CA signed the certificate, the node will be recognized as a registered OpenSSH node. `UserKeyAuth` will lookup the node's details and preform an RBAC check.
+
+### Registering nodes
+
+A new sub-kind to the `node` resource will be added for registered OpenSSH nodes: `openssh`. The absence of a `node` resource sub-kind will imply that a node is a Teleport agent node, making this change backwards compatible.
```

Comment:
Do we still need the new subkind?

Reply:
Yes for Option 2, I guess we could get away with not needing it for Option 1 but it definitely would make debugging easier.